File: iosApp/iosApp/designsystem/data/Dimensions.swift
import Foundation
import shared

public class Dimensions {
    let xTiny: CGFloat
    let tiny: CGFloat
    let small: CGFloat
    let medium: CGFloat
    let large: CGFloat
    let xLarge: CGFloat
    let xxLarge: CGFloat
    let xxxLarge: CGFloat

    init(_ sharedDimension: shared.Dimensions) {
        xTiny = sharedDimension.xTiny.cgFloat
        tiny = sharedDimension.tiny.cgFloat
        small = sharedDimension.small.cgFloat
        medium = sharedDimension.medium.cgFloat
        large = sharedDimension.large.cgFloat
        xLarge = sharedDimension.xLarge.cgFloat
        xxLarge = sharedDimension.xxLarge.cgFloat
        xxxLarge = sharedDimension.xxxLarge.cgFloat
    }

    init(
        xTiny: Int,
        tiny: Int,
        small: Int,
        medium: Int,
        large: Int,
        xLarge: Int,
        xxLarge: Int,
        xxxLarge: Int
    ) {
        self.xTiny = CGFloat(xTiny)
        self.tiny = CGFloat(tiny)
        self.small = CGFloat(small)
        self.medium = CGFloat(medium)
        self.large = CGFloat(large)
        self.xLarge = CGFloat(xLarge)
        self.xxLarge = CGFloat(xxLarge)
        self.xxxLarge = CGFloat(xxxLarge)
    }
}

File: buildSrc/src/main/kotlin/de/stefan/lang/designsystem/dimension/Dimensions.kt
package de.stefan.lang.designsystem.dimension

import de.stefan.lang.designsystem.core.PropertyReader
import de.stefan.lang.designsystem.shapes.Shape
import org.gradle.kotlin.dsl.provideDelegate

data class Dimensions(
    val xTiny: Int,
    val tiny: Int,
    val small: Int,
    val medium: Int,
    val large: Int,
    val xLarge: Int,
    val xxLarge: Int,
    val xxxLarge: Int,
) {
    val all by lazy  { PropertyReader.read<Int, Dimensions>(this) }

    val allSorted by lazy {
        all
            .entries
            .sortedWith (
                compareBy<Map.Entry<String, Int>> { it.value }
            )
            .associate { it.key to it.value }
    }
}
